tests/unit/test_credentials.py

```diff
@@ -211,6 +211,29 @@ def test_can_override_partial_env_var_mapping(self):
         self.assertEqual(creds.secret_key, 'bar')
         self.assertEqual(creds.token, 'baz')
 
+    def test_can_override_expiry_env_var_mapping(self):
```

Comment:
One more test that might be good to add is one that checks that it does not refresh the credentials when it is not going to expire any time soon.